### ISSUE TITLE
plans to separate check with token operation?

### ISSUE THREAD

**Issue by lyang24** [antifuchs/governor#152]
Hi we have an use case to perform check and and token operation independently. for an example when a request come in we check if the remaining quota is enough if so we let request hit db and get the actual amount of token base on db response and finally we try to take n token out?

is the above scenario supported?

**Comment by praetp** [comment]
Is your point that you do not know up front how heavy the database operation will be ?

**Comment by lyang24** [comment]
yes exactly I want to use a billing mode calculate how many tokens after we ran the requests.
